suppression-backup-service/exporter/exporter.go

```diff
@@ -0,0 +1,235 @@
+package exporter
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"path"
+	"time"
+
+	"github.com/rudderlabs/rudder-server/config"
+	suppression "github.com/rudderlabs/rudder-server/enterprise/suppress-user"
+	"github.com/rudderlabs/rudder-server/services/controlplane/identity"
+	"github.com/rudderlabs/rudder-server/services/stats"
+	"github.com/rudderlabs/rudder-server/suppression-backup-service/model"
+	"github.com/rudderlabs/rudder-server/utils/logger"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+)
+
+// ExporterOpt represents a configuration option for the exporter
+type ExporterOpt func(*Exporter)
+
+func Export(repo suppression.Repository, file model.File) error {
+	// export initially to a temp file
+	tmpDir, err := misc.CreateTMPDIR()
+	if err != nil {
+		return fmt.Errorf("could not create tmp dir: %w", err)
+	}
+
+	tempFile, err := os.CreateTemp(tmpDir, "tmp-export")
+	if err != nil {
+		return fmt.Errorf("error creating temp file: %w", err)
+	}
+	defer func() {
+		_ = tempFile.Close()
+		_ = os.Remove(tempFile.Name())
+	}()
+
+	// export to temp file
+	err = repo.Backup(tempFile)
+	if err != nil {
+		return fmt.Errorf("error exporting suppression list: %w", err)
+	}
+
+	// move temp file to final destination
+	file.Mu.Lock()
+	defer file.Mu.Unlock()
+	fmt.Printf("renaming file: %s to %s\n", tempFile.Name(), file.Path)
+	err = os.Rename(tempFile.Name(), file.Path)
+	if err != nil {
+		return fmt.Errorf("error moving temp file to final destination: %w", err)
+	}
+
+	return nil
+}
+
+func newBadgerDBInstance(baseDir string, pkgLogger logger.Logger) (suppression.Repository, error) {
+	err := os.MkdirAll(baseDir, os.ModePerm)
+	if err != nil {
+		return nil, fmt.Errorf("could not create tmp dir: %w", err)
+	}
```

Comment:
BadgerDB creates directories whenever necessary with the appropriate permissions. Do we need this?
```suggestion
	err := os.MkdirAll(baseDir, os.ModePerm)
	if err != nil {
		return nil, fmt.Errorf("could not create tmp dir: %w", err)
	}
```

Reply:
okay, I didn't know that badgerdb does that. Removed it.